core/src/main/java/hudson/slaves/RetentionStrategy.java

```diff
@@ -193,10 +199,25 @@ public String getDisplayName() {
          */
         private final long idleDelay;
 
+        /**
+         * Optional regex (or string in trivial case) with name(s) of node(s)
+         * whose being online blocks the current node from starting.
+         * To properly take advantage of this, be sure to set low idleDelay
+         * timeout values for other workers that no longer have work pending.
+         * Note that conflict is evaluated one way (should this node start
+         * if inDemandDelay has elapsed?) so all nodes in question should
+         * declare each other as conflicting.
+         * This allows a Jenkins deployment to co-locate various pre-defined
+         * or otherwise provisioned build environments without overwhelming
+         * their host - and instead to permit round-robining, one by one.
+         */
+        private final String conflictsWith;
+
         @DataBoundConstructor
-        public Demand(long inDemandDelay, long idleDelay) {
+        public Demand(long inDemandDelay, long idleDelay, String conflictsWith) {
```

Comment:
generally optional fields should be created by a setter (and annotated by `@DataboundSetter`

Reply:
fixed